lib/puppet/provider/zone/solaris.rb

```diff
@@ -263,7 +263,8 @@ def start
 
       unless Puppet::FileSystem.exist?(sysidcfg)
         begin
-          File.open(sysidcfg, "w", 0600) do |f|
+          # For compatibility reasons use System encoding for this OS file
+          Puppet::FileSystem.open(sysidcfg, "w:#{Encoding.default_external.name}", 0600) do |f|
```

Comment:
Interesting - did you look at a Solaris machine... or did you find docs specifying it somewhere?

I was basically trying to mimic what we were already doing - but if we have docs that say this is always UTF-8, happy to switch.

I should also change `Encoding.default_external` to `Encoding.find('filesystem')`, whose existence I only recently learned of if we're not doing `Encoding::UTF_8`